src/main/resources/avro/alleleAnnotationmethods.avdl

```diff
@@ -0,0 +1,205 @@
+@namespace("org.ga4gh.methods")
+protocol AlleleAnnotationMethods {
+
+/*
+This protocol defines methods use to mine pre-calculated variant allele
+annotations.
+*/
+
+import idl "methods.avdl";
+import idl "variants.avdl";
+import idl "alleleAnnotations.avdl";
+import idl "metadata.avdl";
+/*
+variantannotations/search returns annotation for the alleles of Variants
+
+This allows the mining of allele-specific annotations on a VariantSet by
+either a region or by a set of genomic features. Where a region is supplied
+annotation of all alleles vs all features in the region is returned. Where a
+set of features is supplied, only annotations against these features (matching
+on featuretype and id) are returned and other overlapping features are ignored.
+
+variantannotationsets/search returns information on the input to the
+annotation. This will be a VariantSet and the reference data and software
+versions used in calculating the annotation.
+It is essential this information is exhaustive.
+*/
+
+/******************  /variantannotations/search  *********************/
+/** 
+This request maps to the body of `POST /variantannotations/search` as JSON
+*/
+record SearchVariantAnnotationsRequest {
+  /** Required. The ID of the variant annotation set to search over. */
+  string variantAnnotationSetId;
+
+  /**
+  Only return annotation for variants which have exactly this name
+  (case-sensitive, exact match).
+  */
+  union { null, string } name = null;
+
+  /**
+  Only return variants with reference alleles on the reference with this
+  name. One of this field or `referenceId` or `features` is required.
+  (case-sensitive, exact match)
+  */
+  union { null, string } referenceName = null;
+
+  /**
+  Only return variants with reference alleles on the reference with this
+  ID. One of this field or `referenceName` or `features` is required.
+  */
+  union { null, string } referenceId = null;
+
+  /**
+  Required if referenceName or referenceId supplied.
+  The beginning of the window (0-based, inclusive) for which variants with
+  overlapping reference alleles should be returned.
+  Genomic positions are non-negative integers less than reference length.
+  Requests spanning the join of circular genomes are represented as
+  two requests one on each side of the join (position 0).
+  */
+  long start;
+
+  /**
+  Required if referenceName or referenceId supplied.
+  The end of the window (0-based, exclusive) for which variants with
+  overlapping reference alleles should be returned.
+  */
+  long end;
+
+  /**
+  Only return variant annotations for any of these features
+  Features may include specific transcripts, genes or regulatory elements
+  This or a location (referenceName/referenceId plus optional start and end)
+  must be supplied.
+  If null, return all variant annotations in specified window.
+  */
+  union { null, array<string> } featureIds = null;
```

Comment:
@sarahhunt As far as I know `featureIds` are taken verbatim from a variant annotation file and are not GA4GH generated, is this correct? We don't have a taxonomy of features to draw from other than the raw data.

@diekhans could you expand on what does and does not work with this approach? May we suffice to say that, no, since we don't have a priori knowledge of a feature taxonomy that we are simply matching strings?